### PR TITLE
Friendly mobs update

### PR DESCRIPTION
**Pull request by corp-0**
### Purpose:
- [x] Tweak chicken mood to streamline it
- [x] Add mob mood descriptions so people can roughly know what's the mood of a mob by examining it.
- [x] Milkable component to obtain milk from happy milkable mobs.
- [x] Feedable mob component so you can feed the mob with food in your hand.
- [x] Retaliate AI for mobs that only attack when attacked.

### Notes:
I patched BodyPartBehavior so MaxDamage is the same as the MaxHealth from the LivingHealthBehavior class, otherwise as noted by @PerfectTangent mobs would never die.

Cows use the regular friendly AI so they look a little hyperactive.

### Changelog:
CL: added examination interaction to get what a mob's mood.
CL: added new generic mob ai that only becomes hostile after being attacked.
CL: added milkable component to milk mobs based on their mood.
CL: added feedable component as a generic way to make mobs able to be fed by hand.
CL: fixed NRE when interacting with chickens with no item in hand.

### CODE REVIEW

**Comment by corp-0** [comment]
wait wtf, why is it adding old commits

**Comment by corp-0** [comment]
there we go

**Comment by NoooneyDude** [comment]
Reference to discussion on milking mood behaviour: https://discord.com/channels/273774715741667329/273774715741667329/804632247649959966

**Comment by corp-0** [comment]
Done requested changes by @NoooneyDude .

File: UnityProject/Assets/Scripts/NPC/AI/Friendly/ChickenAI.cs [hunk 1]
```diff
@@ -47,14 +40,14 @@ protected override void OnSpawnMob()
 
 		protected override void DoRandomAction()
 		{
-			// Check if we're a happy chiken/chick or we alreday laid all possible egs
-			if (mood.Level < happyChicken || currentLaidEggs == maxEggsAmount)
+			// Check if we alreday laid all possible egs
+			if (currentLaidEggs == maxEggsAmount)
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/AI/Friendly/ChickenAI.cs]:
```suggestion
			if (currentLaidEggs >= maxEggsAmount)
```

Just a small thing, but a safety in consideration for VVing. Up to you. Harmless either way I think.

File: UnityProject/Assets/Scripts/NPC/Mood/MilkableMob.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		private Reagent milkReagent = default;
+
+		[BoxGroup("Reagent")]
+		[SerializeField]
+		[Tooltip("Amount of reagent you get with each interaction.")]
+		private float milkAmount = 25;
+
+		[BoxGroup("Reagent")]
+		[SerializeField]
+		[Tooltip("Max amount of milk the mob can produce before it starts losing mood.")]
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/Mood/MilkableMob.cs]:
```suggestion
		[Tooltip("The update frequency this mob will use for milk production.")]
```

File: UnityProject/Assets/Scripts/NPC/Mood/MilkableMob.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		private float milkTick = 200;
+
+		[BoxGroup("Reagent")]
+		[SerializeField]
+		[Tooltip("Max amount of milk the mob can produce before it starts losing mood.")]
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/NPC/Mood/MilkableMob.cs]:
```suggestion
		[Tooltip("The amount of milk the mob can produce each update period.")]
```

**Comment by corp-0** [comment]
Commited the suggestion

**Pull request merged**